elements/lisk-cryptography/src/encrypt.ts

```diff
@@ -166,3 +169,23 @@ const decryptAES256GCMWithPassword = (
 export const encryptPassphraseWithPassword = encryptAES256GCMWithPassword;
 
 export const decryptPassphraseWithPassword = decryptAES256GCMWithPassword;
+
+export const getKeyPairFromPhraseAndPath = async (phrase: string, path: string) => {
+	if (!isValidPath(path)) {
+		throw new Error('Invalid path format');
+	}
+
+	const masterSeed = await Mnemonic.mnemonicToSeed(phrase);
+	let node = getMasterKeyFromSeed(masterSeed);
+
+	// slice first element which is `m`
+	for (const segment of path.split('/').slice(1)) {
+		// if segment includes apostrophe, we must add HARDENED_OFFSET
+		const segmentWithOffset = segment.includes("'")
```

Comment:
it seems like format logic is in both `isValidPath` and here.
Maybe better to have 
`getPathArray` or something and do validation and conversion to number?

Reply:
I prefer to keep it semantically different, because validate function validates and a get function gets. But if you feel strongly I can update it :)

Reply:
how about `parseKeyDerivationPath` function and throw `parse error`?
Also, this logic will be used in the other BLS key derivation